simavr/sim/sim_gdb.c

```diff
@@ -655,6 +655,10 @@ avr_gdb_init(
 	avr->sleep = avr_callback_sleep_gdb;
 	
 	return 0;
+
+	error:
+		free(g);
```

Comment:
I think it's good idea before calling free(g) at line 660 also close socket, if it was opened, something like this: 
error:
if (g->listen >= 0) close(g->listen);
free(g);